Issue #5 add 2 fa

manrajg13 opened the pull request:
Adding 2FA for users using SMS.

manrajg13 committed:
Add 2FA

manrajg13 committed:
Delete .env

manrajg13 committed:
Update Dockerfile

prairir commented on Dockerfile:
```diff
@@ -5,6 +5,7 @@ RUN useradd microblog
 WORKDIR /home/microblog
 
 COPY requirements.txt requirements.txt
+COPY .env .env
```
move this line to after line 11 and before line 13. That way changing the `.env` file doesnt cause cache invalidation to the steps after.

manrajg13 replied:
Sounds good, I'll move it.

prairir commented on app/auth/forms.py:
```diff
@@ -4,14 +4,33 @@
+        try:
+            return
+        except:
+            print("An exception occurred")
```
1. if you capture an error, you should print out what it should be.
2. this function literally does nothing so well it shouldnt exist.

manrajg13 replied:
It used to have functionality but due to being unable to import the phonenumbers package, I removed some lines for testing and forgot to remove the try except entirely. I'll remove it.

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+            request_verification_token(user.verification_phone)
+            session['username'] = user.username
+            session['phone'] = user.verification_phone
+            return redirect(url_for(
```
if `next` is None or '', you will run into a weird error. The way to fix it is by doing the `next` page check earlier.

manrajg13 replied:
Will do.

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+    if form.validate_on_submit():
+        phone = session['phone']
+        if check_verification_token(phone, form.token.data):
+            del session['phone']
```
I cant figure out why we would need to delete this

manrajg13 replied:
I don't either, I'll remove the deletions.

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+                return redirect(url_for('main.index'))
+            else:
+                username = session['username']
+                del session['username']
```
same as before

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+                username = session['username']
+                del session['username']
+                user = User.query.filter_by(username=username).first()
+                next_page = request.args.get('next')
```
again, if `next` is None of '', you will have an issue. You need to check it

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
 
 
+@bp.route('/verify2fa', methods=['GET', 'POST'])
+def verify_2fa():
```
I cant find where this function is called. There are no requests to it.

Why does this function support both GET and POST? if its a form it should only be POST or PUT.

manrajg13 replied:
The login function uses the verify_2fa function but the route name itself isn't referenced anywhere directly. It's used to map the url after calling its associated function in redirect calls however.

manrajg13 replied:
It gets the form's html template so I believe having a GET method is fine here and in other routes. Though I agree it may not be the best way to do things.

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+    return render_template('auth/verify_2fa.html', form=form)
+
+
+@bp.route('/enable_2fa', methods=['GET', 'POST'])
```
same thing about HTTP methods

prairir commented on app/auth/routes.py:
```diff
@@ -20,14 +21,68 @@ def login():
+    return render_template('auth/enable_2fa.html', form=form)
+
+
+@bp.route('/disable_2fa', methods=['GET', 'POST'])
```
same thing about HTTP methods

prairir reviewed (changes requested):
Overall I think this is great work, I just have a few questions and things that should be changed

manrajg13 committed:
Fix: Dockerfile and routes error handling

Moved copying .env in dockerfile to avoid cache invalidation. Removed
try and except in confirm2fa because it did nothing. Checking if next is
None or '' before returning and removed data deletions in route
functions.

manrajg13 committed:
Merge branch 'main' into Issue-#5-Add-2FA

prairir reviewed (approved):
LGTM :rocket::rocket::rocket::rocket::rocket:
